source/aws_profile.c

```diff
@@ -598,20 +604,35 @@ void aws_profile_collection_destroy(struct aws_profile_collection *profile_colle
 }
 
 static void s_aws_profile_collection_destroy_internal(struct aws_profile_collection *profile_collection) {
-    aws_hash_table_clean_up(&profile_collection->profiles);
+    for (int i = 0; i < AWS_PROFILE_SECTION_TYPE_COUNT; i++) {
+        aws_hash_table_clean_up(&profile_collection->sections[i]);
+    }
     aws_mem_release(profile_collection->allocator, profile_collection);
 }
 
+AWS_STATIC_STRING_FROM_LITERAL(s_profile_token, "profile");
+AWS_STATIC_STRING_FROM_LITERAL(s_sso_session_token, "sso-session");
+
 const struct aws_profile *aws_profile_collection_get_profile(
     const struct aws_profile_collection *profile_collection,
     const struct aws_string *profile_name) {
     struct aws_hash_element *element = NULL;
-    aws_hash_table_find(&profile_collection->profiles, profile_name, &element);
-
+    aws_hash_table_find(&profile_collection->sections[AWS_PROFILE_SECTION_TYPE_PROFILE], profile_name, &element);
     if (element == NULL) {
         return NULL;
     }
+    return element->value;
```

Comment:
triviai: just use the new generic function
```suggestion
    return aws_profile_collection_get_section(profile_collection, AWS_PROFILE_SECTION_TYPE_PROFILE, profile_name);
```